inference-explainers/inference_api.md

```diff
@@ -70,6 +74,58 @@ The optional `warm_up_batch_request_json` is a JSON batch request used to warm u
 
 Refer to the [B&A Inference Onboarding Guide][4] for more details about using the model configuration file with the Terraform configuration.
 
+Note that after a model is loaded, its original metadata entry must be retained to keep the model. Removing the metadata will be treated as a model deletion, as explained in the next section.
+
+### Model Deletion
+
+Ad techs may want to delete stale models to free up memory for the inference sidecar. This can be accomplished by removing stale model entries from the model configuration file. During the periodic fetching process, the inference service checks whether any loaded model is absent from the configuration file. If such models are found, they are deleted from the inference sidecar.
+
+For example, consider the inference service receiving its first model configuration file at time A, which instructs it to load “pcvr_v1” into the inference sidecar. At time B, if “pcvr_v1” is no longer needed and “pcvr_v2” is required instead, Ad techs can upload a new model configuration file to the cloud bucket. This operation will remove “pcvr_v1” from internal storage (since it is no longer listed in the configuration file) and attempt to load “pcvr_v2” from the cloud bucket. In other words, the inference sidecar synchronizes its internal model storage with the updated configuration file.
+
+**Time A**
+
+```json
+{
+  "model_metadata": [
+    {
+      "model_path": "pcvr_v1/",
+      "checksum": "..."
+    }
+  ]
+}
+```
+
+**Time B**
+
+```json
+{
+  "model_metadata": [
+    {
+      "model_path": "pcvr_v2/",
+      "checksum": "..."
+    }
+  ]
+}
+```
+
+When switching model versions, any remaining traffic for the previous model version may need to be processed before fully transitioning to the new version. To avoid traffic disruptions during this process, the inference service can postpone the removal of the older model version. This is achieved by configuring the `eviction_grace_period_in_ms` parameter in the configuration file, which specifies the delay (in milliseconds) before the old model is evicted. For instance, assume that at time A, the following configuration is supplied, upon deletion, the “pcvr_v1” model will be retained for an additional 60 seconds even though “pcvr_v2” is loaded into the inference service. During this grace period, traffic directed to “pcvr_v1” will still be processed.
+
+```json
+{
+  "model_metadata": [
+    {
+      "model_path": "pcvr_v1/",
+      "checksum": "...",
+      "eviction_grace_period_in_ms": 60000,
+    },
+  ]
+}
+```
+
+#### In-Place Update
```

Comment:
Can we make it "### Model In-Place Update" ?

I think CRUD (Loading, Deletion, and Update) functions should be described at the same level.